### PR TITLE
Implement `delete_trace_tag` client API

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/11670?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11670/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11670
```

</p>
</details>

### What changes are proposed in this pull request?

Add `delete_trace_tag` client API.

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

Tested with staging workspace where the backend change is deployed.

```
client = mlflow.MlflowClient()

client.set_trace_tag(request_id=request_id, key="best-onigiri", value="shake")

print(client.get_trace(request_id=request_id).trace_info.tags)
# >> {'best-origiri': 'shake', 'mlflow.user': ...}

# This errors out with 500 response due to a known backend issue, whose fix is already merged but not deployed to the workspace yet. The error is raised from the post RPC validation hook but the tag deletion itself is done properly, so we can still validate the client API.
try:
    client.delete_trace_tag(request_id=backend_request_id, key="best-onigiri")
except MlflowException:
    pass

print(client.get_trace(request_id=request_id).trace_info.tags)
# >> {'mlflow.user': ...}
```

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

- [ ] Yes
- [x] No

### CODE REVIEW

File: mlflow/tracking/client.py [hunk 1]
```diff
@@ -832,13 +832,14 @@ def set_trace_tag(self, request_id: str, key: str, value: str):
 
         Args:
             request_id: The ID of the trace to set the tag on.
-            key: The string key of the tag. Must be shorter than 250 characters.
-            value: The string value of the tag. Must be shorter than 250 characters.
+            key: The string key of the tag. If than 250 characters, otherwise
+                it will be truncated when stored.
+            value: The string value of the tag. Must be shorter than 250 characters, otherwise
+                it will be truncated when stored.
```
Comment (harupy) [inline mlflow/tracking/client.py:838]: nit: `shorter than 250` sounds like the limit is 249. The actual limit is 250, right?

Comment (B-Step62) [reply mlflow/tracking/client.py:838]: Oh true, great catch!

**Review by daniellok-db — APPROVED** [review]
nit: `best-onigiri` value should be `tsuna-mayo`, otherwise lgtm!

**Review by harupy — APPROVED** [review]
Looks good to me!

File: mlflow/tracking/client.py [hunk 2]
```diff
@@ -847,6 +848,40 @@ def set_trace_tag(self, request_id: str, key: str, value: str):
         # If the trace is not active, try to set the tag on the trace in the backend
         self._tracking_client.set_trace_tag(request_id, key, value)
 
+    def delete_trace_tag(self, request_id: str, key: str) -> None:
+        """
+        Delete a tag on the trace with the given trace ID.
+
+        The trace can be an active one or the one that has already ended and recorded in the
+        backend. Below is an example of deleting a tag on an active trace. You can replace the
+        ``request_id`` parameter to setting a tag on an already ended trace.
```
Comment (harupy) [inline mlflow/tracking/client.py:857]:
```suggestion
        backend. Below is an example of deleting a tag on an active trace. You can replace the
        ``request_id`` parameter to deleting a tag on an already ended trace.
```

Should we use `deleting` here?

Comment (harupy) [reply mlflow/tracking/client.py:857]: Should we use `delete` instead of `deleting`?

**Pull request merged**